docker/docker-compose-local.yml

```diff
@@ -1,45 +1,138 @@
 # https://docs.docker.com/compose/compose-file/
 
+x-service-healthy: &service_healthy
+  condition: service_healthy
+
 services:
+  app:
+    build:
+      context: ..
+      dockerfile: docker/app/Dockerfile
+      args:
+        RUBY_VERSION: ${RUBY_VERSION}
+    restart: unless-stopped
+    env_file:
+      - ../.env
+    environment:
+      RAILS_ENV: development
+      RACK_ENV: development
+      NODE_ENV: development
+
+      DATABASE_HOST: db
+      DATABASE_NAME: gumroad_development
+      DATABASE_PORT: 3306
+      DATABASE_USERNAME: root
+      DATABASE_PASSWORD: password
+      SKIP_NATIVE_MYSQL_RECONNECT: "true"
+
+      REDIS_HOST: redis:6379/0
+      SIDEKIQ_REDIS_HOST: redis:6379/1
+      RPUSH_REDIS_HOST: redis:6379/2
+      RACK_ATTACK_REDIS_HOST: redis:6379/3
+      REDIS_URL: redis://redis:6379/0
+      ANYCABLE_REDIS_URL: redis://redis:6379/5
+
+      MONGO_DATABASE_URL: mongo:27017
+      MONGO_DATABASE_NAME: gumroad_log_development
+      MONGO_DATABASE_USERNAME: ""
+      MONGO_DATABASE_PASSWORD: ""
+
+      ELASTICSEARCH_HOST: http://elasticsearch:9200
+      MEMCACHE_SERVERS: memcached:11211
+
+      ANYCABLE_HOST: 0.0.0.0
+      SHAKAPACKER_DEV_SERVER_HOST: 0.0.0.0
+    volumes:
+      - ..:/workspace
+      - bundle:/bundle_path
+      - node_modules:/workspace/node_modules
+    depends_on:
+      db: *service_healthy
+      redis: *service_healthy
+      mongo: *service_healthy
+      elasticsearch: *service_healthy
+      memcached: *service_healthy
+      minio: *service_healthy
+    tty: true
+    stdin_open: true
+    healthcheck:
```

Comment:
I added a health check for each service so they start in the correct order and it’s easier to see when something goes wrong.

<img width="1932" height="423" alt="Image" src="https://github.com/user-attachments/assets/49a28c55-4695-4f08-92f8-3aef5fa5f83e" />